waiter/src/waiter/async_request.clj

```diff
@@ -225,7 +262,9 @@
    {:keys [request-id] :as reason-map} request-properties location query-string]
   (let [correlation-id (cid/get-correlation-id)
         {:strs [metric-group backend-proto]} service-description
-        status-endpoint (scheduler/end-point-url backend-proto host port location)
+        request-proto (get-async-request-protocol instance port service-description)
+        proto (when (not= request-proto backend-proto) request-proto)
```

Comment:
ℹ️ Since the option for the Scheduler to pick a different request protocol is coming in a follow-up patch, there isn't a straightforward way to integration test it in this PR. To help verify that the `/waiter-async/v2/` api actually works, I manually ran the `waiter.async-request-integration-test` integration tests with the logic in this line inverted.